pkg/cmd/flags/policy.go

```diff
@@ -0,0 +1,291 @@
+package flags
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/events"
+)
+
+// PolicyFile is the structure of the policy file
+type PolicyFile struct {
+	Name          string   `yaml:"name"`
+	Description   string   `yaml:"description"`
+	Scope         []string `yaml:"scope"`
+	DefaultAction string   `yaml:"defaultAction"`
+	Rules         []Rule   `yaml:"rules"`
+}
+
+// Rule is the structure of the rule in the policy file
+type Rule struct {
+	Event  string   `yaml:"event"`
+	Filter []string `yaml:"filter"`
+	Action string   `yaml:"action"`
+}
+
+// PrepareFilterMapForPolicies prepares the FilterMap for the policies
+func PrepareFilterMapFromPolicies(policies []PolicyFile) (FilterMap, error) {
+	filterMap := make(FilterMap)
+
+	if len(policies) == 0 {
+		return nil, errfmt.Errorf("no policies provided")
+	}
+
+	if len(policies) > 64 {
+		return nil, errfmt.Errorf("too many policies provided, there is a limit of 64 policies")
+	}
+
+	for i, p := range policies {
+		err := validatePolicy(p)
+		if err != nil {
+			return nil, err
+		}
+
+		filterFlags := make([]*filterFlag, 0)
+
+		err = validateAction(p.Name, p.DefaultAction)
+		if err != nil {
+			return nil, err
+		}
+
+		// scope
+		for _, s := range p.Scope {
+			s = strings.ReplaceAll(s, " ", "")
+
+			if s == "global" && len(p.Scope) > 1 {
+				return nil, errfmt.Errorf("policy %s, global scope must be unique", p.Name)
+			}
+
+			if s == "global" {
+				break
+			}
+
+			var scope, filterName, operatorAndValues string
+
+			switch s {
+			case "follow", "!container":
+				scope = s
+				filterName = s
+				operatorAndValues = ""
+			default:
+				operatorIdx := strings.IndexAny(s, "=!<>")
+
+				if operatorIdx == -1 {
+					return nil, errfmt.Errorf("policy %s, scope %s is not valid", p.Name, s)
+				}
+
+				scope = s[:operatorIdx]
+				filterName = s[:operatorIdx]
+				operatorAndValues = s[operatorIdx:]
+			}
+
+			err := validateScope(p.Name, scope)
+			if err != nil {
+				return nil, err
+			}
+
+			filterFlags = append(filterFlags, &filterFlag{
+				full:              s,
+				filterName:        filterName,
+				operatorAndValues: operatorAndValues,
+				policyIdx:         i,
+			})
+		}
+
+		events := make(map[string]bool)
+
+		for _, r := range p.Rules {
+			err := validateEvent(p.Name, r.Event)
+			if err != nil {
+				return nil, err
+			}
+
+			if _, ok := events[r.Event]; ok {
+				return nil, errfmt.Errorf("policy %s, event %s is duplicated", p.Name, r.Event)
```

Comment:
we can add a note that currently only one rule per event is supported (we may add support for more than one rule per event in the future)

Reply:
Actually I meant that the error message should indicate that, something like:
"policy x uses event y in more than one rule which is not currently supported"

Reply:
AHHHHH! sorry, fixing it